src/main/java/io/buji/pac4j/subject/Pac4jPrincipal.java

```diff
@@ -70,4 +71,16 @@ public boolean equals(Object o) {
     public int hashCode() {
         return profiles != null ? profiles.hashCode() : 0;
     }
+
+	@Override
+	public String getName() {
+		CommonProfile profile = this.getProfile();
+		return profile.getId();
+		
+	}
+	
+	@Override
+	public String toString() {
```

Comment:
In fact, I would return all profiles. Something like: `CommonHelper.toString(this.getClass(), "profiles", getPRofiles());`